### PR TITLE
Update and add test to validate energy returned in the solution is corre...

### PR DESCRIPTION
**Pull request by hemantdwave**
Please review changes done to validate energy returned in the solution

### CODE REVIEW

File: tests/test_knapsack.py [hunk 1]
```diff
@@ -45,13 +45,44 @@ def test_knapsack(self):
...
-                                         data_file_path])
+                                         data_file_path, "50"])
         output = str(output).upper()
         if os.getenv('DEBUG_OUTPUT'):
             print("Example output :-"+ output)
 
         with self.subTest(msg="Verify if output contains 'FOUND SOLUTION' \n"):
             self.assertIn("FOUND SOLUTION", output)
+        with self.subTest(msg="Verify if output contains correct energy' \n"):
+            self.assertIn("ENERGY -205.0", output)
```
Comment (randomir) [inline tests/test_knapsack.py:56]: Is this robustly reproducible? From @joelgdwave's comments sounds like it isn't.

One way to handle the heuristic nature of solver/solution is to retry a few times, if energy doesn't match the expected one.

Another way is to use a less strict assertion, e.g. assert reasonable bounds only.

File: tests/test_knapsack.py [hunk 2]
```diff
@@ -45,13 +45,44 @@ def test_knapsack(self):
...
+        
+        # Get energy value returned by solution
+        index = output.index("ENERGY") + 7
+        energy = ""
+        i = 0
+        while i < 5 and len(output)> index:
+            energy = energy + output[index]
+            i = i + 1
+            index = index + 1
+        energy = (float(energy))
```
Comment (randomir) [inline tests/test_knapsack.py:80]:
```suggestion
        import re
        regex = r'energy\s*([+-]?\d+(\.\d*)?)'
        energy = float(next(re.finditer(regex, output, re.I)).group(1))
```

File: tests/test_knapsack.py [hunk 3]
```diff
@@ -45,13 +45,44 @@ def test_knapsack(self):
...
+        
+        with self.subTest(msg="Verify if output contains 'FOUND SOLUTION' \n"):
+            self.assertIn("FOUND SOLUTION", output)
+        with self.subTest(msg="Verify if the energy needs to be between -519 and -524' \n"):
```
Comment (randomir) [inline tests/test_knapsack.py:84]:
```suggestion
        with self.subTest(msg="Verify energy is between -519 and -524' \n"):
```

**Comment by joelgdwave** [comment]
For HSS v2, Will had found a bug, which hasn't been pushed out yet.
Once HSS is updated, the -205, and -519 to -524 should happen much more often.

I agree with the re-try and check again approach.

**Pull request merged**